File: backend/internal/models/hashtag.go
package models

import "time"

type Hashtag struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	Slug      string    `json:"slug"`
	Category  string    `json:"category"` // New field
	CreatedBy string    `json:"createdBy"`
	CreatedAt time.Time `json:"createdAt"`
	Followers int       `json:"followers"`
}

type HashtagPost struct {
	HashtagID string    `json:"hashtagId"`
	PostID    string    `json:"postId"`
	IsBoost   bool      `json:"isBoost"` // true for boost, false for shout
	CreatedAt time.Time `json:"createdAt"`
}

File: packages/shared/src/domain/hashtag.ts
/**
 * Hashtag Domain Models
 * 
 * Types and interfaces for hashtags.
 */

import { Post } from './post';

export interface Hashtag {
  id: string;
  name: string;
  slug: string;
  category?: string;
  createdBy: string;
  postCount?: number;
  createdAt: Date | string;
}

export interface HashtagPost {
  hashtagId: string;
  postId: string;
  isBoost: boolean; // true for boost, false for shout
  createdAt: Date | string;
}

export interface CreateHashtagRequest {
  name: string;
  slug?: string;
  category?: string;
  createdBy: string;
}

export interface HashtagListParams {
  limit?: number;
  offset?: number;
  search?: string;
}
